include/revng/ADT/Iterator.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <iterator>
+#include <type_traits>
+
+#include "llvm/ADT/STLExtras.h"
+
+auto derefereceIterator(auto Iter) {
+  return llvm::map_iterator(Iter, [](const auto &Ptr) -> decltype(*Ptr) & {
+    return *Ptr;
+  });
+}
+
+auto derefereceRange(auto &&Range) {
```

Comment:
Ibidem.

Perfect forwarding maybe?

Reply:
we can't because we are digging into begin and end of the range